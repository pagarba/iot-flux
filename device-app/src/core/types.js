export default {
  LOGIN_REQUEST: 'LOGIN_REQUEST',
  LOGIN_SUCCESS: 'LOGIN_SUCCESS',
  LOGIN_FAILURE: 'LOGIN_FAILURE',
  LOGOUT_SUCCESS: 'LOGOUT_SUCCESS',
  LOGOUT_FAILURE: 'LOGOUT_FAILURE',
  ACCOUNT_FETCH_SUCCESS: 'ACCOUNT_FETCH_SUCCESS',

  CREATE_USER_REQUEST: 'CREATE_USER_REQUEST',
  CREATE_USER_SUCCESS: 'CREATE_USER_SUCCESS',
  CREATE_USER_FAILURE: 'CREATE_USER_FAILURE',

  CREATE_DEVICE_REQUEST: 'CREATE_DEVICE_REQUEST',
  CREATE_DEVICE_SUCCESS: 'CREATE_DEVICE_SUCCESS',
  CREATE_DEVICE_FAILURE: 'CREATE_DEVICE_FAILURE',

  EDIT_DEVICE_REQUEST: 'EDIT_DEVICE_REQUEST',
  EDIT_DEVICE_SUCCESS: 'EDIT_DEVICE_SUCCESS',
  EDIT_DEVICE_FAILURE: 'EDIT_DEVICE_FAILURE',

  GET_DEVICES_REQUEST: 'GET_DEVICES_REQUEST',
  GET_DEVICES_SUCCESS: 'GET_DEVICES_SUCCESS',
  GET_DEVICES_FAILURE: 'GET_DEVICES_FAILURE',

  GET_DEVICE_REQUEST: 'GET_DEVICE_REQUEST',
  GET_DEVICE_SUCCESS: 'GET_DEVICE_SUCCESS',
  GET_DEVICE_FAILURE: 'GET_DEVICE_FAILURE',

  DELETE_DEVICE_REQUEST: 'DELETE_DEVICE_REQUEST',
  DELETE_DEVICE_SUCCESS: 'DELETE_DEVICE_SUCCESS',
  DELETE_DEVICE_FAILURE: 'DELETE_DEVICE_FAILURE',

  CREATE_CHANNEL_REQUEST: 'CREATE_CHANNEL_REQUEST',
  CREATE_CHANNEL_SUCCESS: 'CREATE_CHANNEL_SUCCESS',
  CREATE_CHANNEL_FAILURE: 'CREATE_CHANNEL_FAILURE',

  EDIT_CHANNEL_REQUEST: 'EDIT_CHANNEL_REQUEST',
  EDIT_CHANNEL_SUCCESS: 'EDIT_CHANNEL_SUCCESS',
  EDIT_CHANNEL_FAILURE: 'EDIT_CHANNEL_FAILURE',

  GET_CHANNELS_REQUEST: 'GET_CHANNELS_REQUEST',
  GET_CHANNELS_SUCCESS: 'GET_CHANNELS_SUCCESS',
  GET_CHANNELS_FAILURE: 'GET_CHANNELS_FAILURE',

  GET_CHANNEL_REQUEST: 'GET_CHANNEL_REQUEST',
  GET_CHANNEL_SUCCESS: 'GET_CHANNEL_SUCCESS',
  GET_CHANNEL_FAILURE: 'GET_CHANNEL_FAILURE',

  DELETE_CHANNEL_REQUEST: 'DELETE_CHANNEL_REQUEST',
  DELETE_CHANNEL_SUCCESS: 'DELETE_CHANNEL_SUCCESS',
  DELETE_CHANNEL_FAILURE: 'DELETE_CHANNEL_FAILURE',

  ADD_DEVICE_TO_CHANNEL_REQUEST: 'ADD_DEVICE_TO_CHANNEL_REQUEST',
  ADD_DEVICE_TO_CHANNEL_SUCCESS: 'ADD_DEVICE_TO_CHANNEL_SUCCESS',
  ADD_DEVICE_TO_CHANNEL_FAILURE: 'ADD_DEVICE_TO_CHANNEL_FAILURE',

  DELETE_DEVICE_FROM_CHANNEL_REQUEST: 'DELETE_DEVICE_FROM_CHANNEL_REQUEST',
  DELETE_DEVICE_FROM_CHANNEL_SUCCESS: 'DELETE_DEVICE_FROM_CHANNEL_SUCCESS',
  DELETE_DEVICE_FROM_CHANNEL_FAILURE: 'DELETE_DEVICE_FROM_CHANNEL_FAILURE',

  GET_CHANNEL_DEVICES_REQUEST: 'GET_CHANNEL_DEVICES_REQUEST',
  GET_CHANNEL_DEVICES_SUCCESS: 'GET_CHANNEL_DEVICES_SUCCESS',
  GET_CHANNEL_DEVICES_FAILURE: 'GET_CHANNEL_DEVICES_FAILURE',

  CREATE_EVENT_REQUEST: 'CREATE_EVENT_REQUEST',
  CREATE_EVENT_SUCCESS: 'CREATE_EVENT_SUCCESS',
  CREATE_EVENT_FAILURE: 'CREATE_EVENT_FAILURE',

  EDIT_EVENT_REQUEST: 'EDIT_EVENT_REQUEST',
  EDIT_EVENT_SUCCESS: 'EDIT_EVENT_SUCCESS',
  EDIT_EVENT_FAILURE: 'EDIT_EVENT_FAILURE',

  GET_EVENTS_REQUEST: 'GET_EVENTS_REQUEST',
  GET_EVENTS_SUCCESS: 'GET_EVENTS_SUCCESS',
  GET_EVENTS_FAILURE: 'GET_EVENTS_FAILURE',

  GET_EVENT_REQUEST: 'GET_EVENT_REQUEST',
  GET_EVENT_SUCCESS: 'GET_EVENT_SUCCESS',
  GET_EVENT_FAILURE: 'GET_EVENT_FAILURE',

  DELETE_EVENT_REQUEST: 'DELETE_EVENT_REQUEST',
  DELETE_EVENT_SUCCESS: 'DELETE_EVENT_SUCCESS',
  DELETE_EVENT_FAILURE: 'DELETE_EVENT_FAILURE',
}