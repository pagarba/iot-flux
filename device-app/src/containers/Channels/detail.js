import React, { Component } from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';

import { connect } from 'react-redux';

import { withStyles } from "@material-ui/core/styles/index";
import {getChannel, editChannel, getChannelDevices} from '../../core/actions/channel'
import ChannelDevices from './Devices';

const styles = theme => ({
  root: {
    display: 'block',
  },
  button: {
    margin: theme.spacing.unit,
    marginTop: '2em',
    color: '#fff',
  },
  heading: {
    fontSize: theme.typography.pxToRem(15),
    flexBasis: '33.33%',
    flexShrink: 0,
    fontWeight: 'bold',
  },
  formControl: {
    margin: theme.spacing.unit,
  },
  secondaryHeading: {
    fontSize: theme.typography.pxToRem(15),
    color: theme.palette.text.secondary,
  },
});

export class ChannelDetail extends Component {
  constructor () {
    super();

    this.state = {
      isEditMode: false,
      name: '',
    };
  }

  componentWillMount() {
    const channelId = this.props.match.params.channelId;

    this.props.getChannel(channelId);
    this.props.getChannelDevices(channelId);
  }

  componentWillReceiveProps(props) {
    this.setState({
      name: props.channel.name,
    })
  }

  editChannel = () => {
    const { name } = this.state;
    const channelId = this.props.match.params.channelId;

    this.props.editChannel(channelId, {
      name
    }).then(() => {
      this.props.getChannel(channelId).then(() => {
        this.setState({
          isEditMode: false,
        })
      });
    });
  };

  handleInputChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  render() {
    const { classes } = this.props;

    const channel = this.props.channel || {};

    return (
      <div className={classes.root}>
        <ChannelDevices
          channel={channel}
        />
      </div>
    )
  }
}

ChannelDetail.propTypes = {
  classes: PropTypes.object.isRequired,
  theme: PropTypes.object.isRequired,
};

function mapStateToProps(state) {
  return {
    channel: state.rootReducer.channel.channel,
    isEditingChannel: state.rootReducer.channel.isEditingChannel,
  }
}

function mapDispatchToProps(dispatch) {
  return {
    getChannel: (id) => dispatch(getChannel(id)),
    editChannel: (id, data) => dispatch(editChannel(id, data)),
    getChannelDevices: (deviceId) => dispatch(getChannelDevices(deviceId)),
  }
}

const WithStyles =  withStyles(styles, { withTheme: true })(ChannelDetail);
export default connect(mapStateToProps, mapDispatchToProps)(WithStyles);
