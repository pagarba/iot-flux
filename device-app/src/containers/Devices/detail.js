import React, { Component } from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';

import { connect } from 'react-redux';

import FormControl from '@material-ui/core/FormControl';
import Input from '@material-ui/core/Input';

import { withStyles } from "@material-ui/core/styles/index";
import { getDevice, editDevice } from '../../core/actions/device';

const styles = theme => ({
  root: {
    display: 'block'
  },
  button: {
    margin: theme.spacing.unit,
    marginTop: '2em',
    color: '#fff',
  },
  heading: {
    fontSize: theme.typography.pxToRem(15),
    flexBasis: '33.33%',
    flexShrink: 0,
    fontWeight: 'bold',
  },
  formControl: {
    margin: theme.spacing.unit,
  },
  secondaryHeading: {
    fontSize: theme.typography.pxToRem(15),
    color: theme.palette.text.secondary,
  },
});

export class DeviceDetail extends Component {
  constructor () {
    super();

    this.state = {
      isEditMode: false,
      key: '',
      name: '',
      type: '',
    };
  }

  componentWillMount() {
    const deviceId = this.props.match.params.deviceId;
    this.props.getDevice(deviceId);
  }

  componentWillReceiveProps(props) {
    this.setState({
      key: props.device.key,
      name: props.device.name,
      type: props.device.type,
    })
  }

  editDevice = () => {
    const { name, type } = this.state;
    const deviceId = this.props.match.params.deviceId;

    this.props.editDevice(deviceId, {
      name,
      type
    }).then(() => {
      this.props.getDevice(deviceId).then(() => {
        this.setState({
          isEditMode: false,
        })
      });
    });
  }

  handleInputChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  render() {
    const { classes } = this.props;
    const { isEditMode } = this.state;

    const device = this.props.device || {};

    return (
      <div className={classes.root}>
        <table>
          <tbody>
            <tr>
              <td><b>Device ID: </b></td>
              <td>
                {device.id}
              </td>
            </tr>
            <tr>
              <td><b>Device Key: </b></td>
              <td>{ device.key }</td>
            </tr>
            <tr>
              <td><b>Device Type: </b></td>
              <td>{device.type}</td>
            </tr>
            <tr>
              <td><b>Device Name: </b></td>
              <td>
                {
                  isEditMode ? (
                    <FormControl className={classes.formControl} error={this.state.name === null} fullWidth aria-describedby="component-error-text">
                      <Input name="name" value={this.state.name} onChange={this.handleInputChange} />
                    </FormControl>
                  ) : device.name
                }
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    )
  }
}

DeviceDetail.propTypes = {
  classes: PropTypes.object.isRequired,
  theme: PropTypes.object.isRequired,
};

function mapStateToProps(state) {
  return {
    device: state.rootReducer.device.device,
    isEditingDevice: state.rootReducer.device.isEditingDevice,
  }
}

function mapDispatchToProps(dispatch) {
  return {
    getDevice: (id) => dispatch(getDevice(id)),
    editDevice: (id, data) => dispatch(editDevice(id, data)),
  }
}

const WithStyles =  withStyles(styles, { withTheme: true })(DeviceDetail);
export default connect(mapStateToProps, mapDispatchToProps)(WithStyles);
