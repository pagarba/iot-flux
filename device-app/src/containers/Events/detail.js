import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux'

import ExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import FormControl from '@material-ui/core/FormControl';
import FormHelperText from '@material-ui/core/FormHelperText';
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';
import InputAdornment from '@material-ui/core/InputAdornment';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';

import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import SearchIcon from '@material-ui/icons/Search';

import { withStyles } from "@material-ui/core/styles/index";

import { getChannel, getChannels, getChannelDevices } from '../../core/actions/channel';
import { createEvent, getEvents, emptyEvents } from '../../core/actions/event';
import { getDevices } from '../../core/actions/device';

import EventsTable from '../../components/Tables/EventsTable';
import CreateConfirmDialog from '../../components/Dialogs/CreateConfirmDialog';
import moment from "moment/moment";

const styles = theme => ({
  root: {
    display: 'block'
  },
  button: {
    margin: theme.spacing.unit,
    marginTop: '2em',
    color: '#fff',
  },
  channelFilter: {
    backgroundColor: '#fff',
    width: '100%',
    display: 'flex',
    alignItems: 'flex-end',
    borderTopLeftRadius: '4px',
    borderTopRightRadius: '4px',
  },
  eventsListPanel: {
    backgroundColor: '#52cee8',
  },
  eventsListPanelDetails: {
    display: 'flex',
    flexDirection: 'column',
  },
  heading: {
    fontSize: theme.typography.pxToRem(15),
    flexShrink: 0,
    fontWeight: 'bold',
  },
  formControl: {
    minWidth: '150px',
    margin: '0 1rem',
  },
  secondaryHeading: {
    fontSize: theme.typography.pxToRem(15),
    color: theme.palette.text.secondary,
  },
});

export class EventsDetail extends Component {
  state = {
    channelId: '',
    expanded: 'panel1',
    eventName: '',
    deviceKey: '',
    data: '',
    isCreateConfirmDialogOpen: false,
    searchKeyword: '',
  };

  async componentDidMount() {

    /**
     * Latest way to get channel devices. (With dropdown and with updated mainflux API)
     */
    this.props.getDevices();
    await this.props.getChannels();

    let { channels } = this.props;
    channels = channels.channels;

    if (channels.length) {
      await this.props.getChannelDevices(channels[0].id);

      this.setState({
        channelId: channels[0].id,
      });

      const { channelDevices } = this.props;
      if (channelDevices && channelDevices.things && channelDevices.things.length) {
        const deviceKey = channelDevices.things[0].key;

        this.props.getEvents(channels[0].id, deviceKey);
      } else {
        this.props.getEvents(channels[0].id, 'random key');
      }
    } else {
      this.props.emptyEvents();
    }
  }

  addEvent = () => {
    const { channelId, deviceKey, data } = this.state;

    if (data && deviceKey) {
      this.props.createEvent(channelId, deviceKey, data)
        .then(() => {
          this.props.getEvents(channelId, deviceKey);
          this.setState({
            deviceKey: '',
            data: '',
            isCreateConfirmDialogOpen: true,
          })
        });
    }
  };

  handleChange = panel => (event, expanded) => {
    this.setState({
      expanded: expanded ? panel : false,
    });
  };

  handleBlur = event => {
    if (!this.state[event.target.name])
      this.setState({ [event.target.name]: null });
  };

  handleHideCreateConfirmDialog = () => {
    this.setState({
      isCreateConfirmDialogOpen: false,
    })
  };

  handleInputChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  handleChannelChange = async (event) => {
    this.setState({
      channelId: event.target.value,
    });

    this.props.getChannel(event.target.value);
    await this.props.getChannelDevices(event.target.value);

    const { channelDevices } = this.props;

    if (channelDevices && channelDevices.things && channelDevices.things.length) {
      const deviceKey = channelDevices.things[0].key;
      this.props.getEvents(event.target.value, deviceKey);
    } else {
      this.props.getEvents(event.target.value, 'random key');
    }
  };

  navigateToView = () => {
    this.setState({
      isCreateConfirmDialogOpen: false,
      expanded: 'panel1'
    });
  };

  navigateToCreate = () => {
    this.setState({
      isCreateConfirmDialogOpen: false,
      expanded: 'panel2'
    });
  };

  filteredMessages = () => {
    const { events, devices } = this.props;
    const { searchKeyword } = this.state;

    return ((events || {}).messages || []).filter((e) => {
      const deviceName = devices && devices.things ? (devices.things.find((device) => device.id === e.publisher) || {}).name : '';
      return e.name.indexOf(searchKeyword) !== -1 || e.protocol.indexOf(searchKeyword) !== -1
        || e.unit.indexOf(searchKeyword) !== -1 || e.value.toString().indexOf(searchKeyword) !== -1
        || moment(e.time).format('YYYY-MM-DD HH:mm:ss').indexOf(searchKeyword) !== -1
        || deviceName.indexOf(searchKeyword) !== -1;
    });
  };

  render() {
    const { classes, history, channels, devices, channelDevices } = this.props;
    const { expanded } = this.state;

    return (
      <div className={classes.root}>
        <ExpansionPanel
          className={classes.eventsListPanel}
          expanded={expanded === 'panel1'}
          onChange={this.handleChange('panel1')}
        >
          <ExpansionPanelSummary expandIcon={<ExpandMoreIcon color="primary" />}>
            <Typography className={classes.heading} color="primary">MY EVENTS</Typography>
          </ExpansionPanelSummary>
          <ExpansionPanelDetails className={classes.eventsListPanelDetails}>
            <div className={classes.channelFilter}>
              <FormControl className={classes.formControl}>
                <InputLabel htmlFor="deviceKey">Select Channel</InputLabel>
                <Select
                  value={this.state.channelId}
                  onChange={this.handleChannelChange}
                  inputProps={{
                    name: 'channelId',
                    id: 'channelId',
                  }}
                >
                  {
                    ((channels || {}).channels || []).map((channel) => (
                      <MenuItem value={channel.id}>{channel.name}</MenuItem>
                    ))
                  }
                </Select>
              </FormControl>
              <FormControl className={classes.formControl}>
                <Input
                  id="searchKeyword"
                  name="searchKeyword"
                  onChange={this.handleInputChange}
                  startAdornment={
                    <InputAdornment position="start">
                      <SearchIcon />
                    </InputAdornment>
                  }
                />
              </FormControl>
            </div>
            <EventsTable
              history={history}
              data={this.filteredMessages()}
              devices={devices}
            />
          </ExpansionPanelDetails>
        </ExpansionPanel>
        <ExpansionPanel expanded={expanded === 'panel2'} onChange={this.handleChange('panel2')}>
          <ExpansionPanelSummary expandIcon={<ExpandMoreIcon />}>
            <Typography className={classes.heading}>ADD A NEW EVENT</Typography>
          </ExpansionPanelSummary>
          <ExpansionPanelDetails className={classes.root}>
            <FormControl className={classes.formControl}>
              <InputLabel htmlFor="deviceKey">Device</InputLabel>
              <Select
                value={this.state.deviceKey}
                onChange={this.handleInputChange}
                inputProps={{
                  name: 'deviceKey',
                  id: 'deviceKey',
                }}
              >
                {
                  channelDevices && channelDevices.things && channelDevices.things.map((device) => (
                    <MenuItem value={device.key}>{device.name}</MenuItem>
                  ))
                }
              </Select>
            </FormControl>

            <FormControl className={classes.formControl} error={this.state.data === null} fullWidth aria-describedby="component-error-text">
              <InputLabel htmlFor="component-error">Data</InputLabel>
              <Input name="data" value={this.state.data} onChange={this.handleInputChange} onBlur={this.handleBlur} />
              <FormHelperText id="component-error-text">This field is required</FormHelperText>
            </FormControl>

            <Button
              className={classes.button}
              color="secondary"
              onClick={this.addEvent}
              variant="contained"
              disabled={this.props.isCreatingEvent}
            >
              Add Event
            </Button>
          </ExpansionPanelDetails>
        </ExpansionPanel>
        <CreateConfirmDialog
          open={this.state.isCreateConfirmDialogOpen}
          onClickDismiss={this.handleHideCreateConfirmDialog}
          onClickView={this.navigateToView}
          onClickCreate={this.navigateToCreate}
          contentText="A new event has been created"
          titleText="Confirm"
          viewButtonText="View Events"
          createButtonText="Create Event"
        />
      </div>
    )
  }
}

EventsDetail.propTypes = {
  classes: PropTypes.object.isRequired,
  theme: PropTypes.object.isRequired,
};

function mapStateToProps(state) {
  return {
    channel: state.rootReducer.channel.channel,
    channels: state.rootReducer.channel.channels,
    channelDevices: state.rootReducer.channel.channelDevices,
    events: state.rootReducer.event.events,
    devices: state.rootReducer.device.devices,
    isCreatingEvent: state.rootReducer.event.isCreatingEvent,
  }
}

function mapDispatchToProps(dispatch) {
  return {
    createEvent: (channelId, deviceKey, data) => dispatch(createEvent(channelId, deviceKey, data)),
    getChannel: (id) => dispatch(getChannel(id)),
    getChannels: () => dispatch(getChannels()),
    getChannelDevices: (id) => dispatch(getChannelDevices(id)),
    getDevices: () => dispatch(getDevices()),
    getEvents: (channelId, deviceKey) => dispatch(getEvents(channelId, deviceKey)),
    emptyEvents: () => dispatch(emptyEvents()),
  }
}

const WithStyles =  withStyles(styles, { withTheme: true })(EventsDetail);
export default connect(mapStateToProps, mapDispatchToProps)(WithStyles);
