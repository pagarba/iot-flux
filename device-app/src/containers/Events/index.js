import React, { Component } from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';

import { connect } from 'react-redux';
import qs from 'stringquery';
import ExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import Typography from '@material-ui/core/Typography';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';

import { withStyles } from "@material-ui/core/styles/index";

import EventChannelsTable from '../../components/Tables/EventChannelsTable';
import { createChannel, getChannels, deleteChannel } from '../../core/actions/channel';

const styles = theme => ({
  root: {
    display: 'block'
  },
  button: {
    margin: theme.spacing.unit,
    marginTop: '2em'
  },
  eventsListPanel: {
    backgroundColor: '#52cee8',
  },
  heading: {
    fontSize: theme.typography.pxToRem(15),
    flexShrink: 0,
    fontWeight: 'bold',
  },
  formControl: {
    margin: theme.spacing.unit,
  },
  secondaryHeading: {
    fontSize: theme.typography.pxToRem(15),
    color: theme.palette.text.secondary,
  },
});

export class Events extends Component {
  state = {
    expanded: null,
    channelName: '',
  };

  componentWillMount() {
    this.props.getChannels();
  }

  componentDidMount() {
    this.setState({
      expanded: 'panel1'
    })
  }

  handleChange = panel => (event, expanded) => {
    this.setState({
      expanded: expanded ? panel : false,
    });
  };

  handleBlur = event => {
    if (!this.state[event.target.name])
      this.setState({ [event.target.name]: null });
  };

  handleInputChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  addChannel = () => {
    const { channelName } = this.state;

    if (channelName) {
      this.props.createChannel({
        name: channelName,
      }).then(() => {
        this.props.getChannels();
      });
    }
  };

  deleteChannel = (id) => {
    this.props.deleteChannel(id)
      .then(() => {
        this.props.getChannels();
      })
      .catch(() => {
        alert('Error while deleting channel');
      });
  }

  render() {
    const { classes, history } = this.props;
    const { expanded } = this.state;

    return (
      <div className={classes.root}>
        <ExpansionPanel
          className={classes.eventsListPanel}
          expanded={expanded === 'panel1'}
          onChange={this.handleChange('panel1')}
        >
          <ExpansionPanelSummary expandIcon={<ExpandMoreIcon color="primary" />}>
            <Typography color="primary" className={classes.heading}>MY CHANNELS</Typography>
          </ExpansionPanelSummary>
          <ExpansionPanelDetails>
            <EventChannelsTable
              history={history}
              data={this.props.channels ? this.props.channels.channels : []}
            />
          </ExpansionPanelDetails>
        </ExpansionPanel>
      </div>
    )
  }
}

Events.propTypes = {
  classes: PropTypes.object.isRequired,
  theme: PropTypes.object.isRequired,
};

function mapStateToProps(state) {
  return {
    channels: state.rootReducer.channel.channels,
    isCreatingDevice: state.rootReducer.device.isCreatingDevice,
  }
}

function mapDispatchToProps(dispatch) {
  return {
    getChannels: () => dispatch(getChannels()),
    createChannel: data => dispatch(createChannel(data)),
    deleteChannel: id => dispatch(deleteChannel(id)),
  }
}

const WithStyles =  withStyles(styles, { withTheme: true })(Events);
export default connect(mapStateToProps, mapDispatchToProps)(WithStyles);