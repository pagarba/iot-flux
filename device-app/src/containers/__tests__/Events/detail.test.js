import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';

import { EventsDetail } from '../../Events/detail';

describe('<EventsDetail />', () => {
  let shallow;

  beforeAll(() => {
    shallow = createShallow();
  });

  it('should render components', () => {
    const wrapper = shallow(
      <EventsDetail
        classes={{}}
        match={{params: {}}}
        getDevices={jest.fn()}
        getChannel={() => new Promise(resolve => resolve())}
        theme={{}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});