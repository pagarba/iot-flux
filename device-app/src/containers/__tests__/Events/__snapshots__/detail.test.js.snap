// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventsDetail /> should render components 1`] = `
<div>
  <WithStyles(ExpansionPanel)
    expanded={true}
    onChange={[Function]}
  >
    <WithStyles(ExpansionPanelSummary)
      expandIcon={
        <pure(ExpandMoreIcon)
          color="primary"
        />
      }
    >
      <WithStyles(Typography)
        color="primary"
      >
        MY EVENTS
      </WithStyles(Typography)>
    </WithStyles(ExpansionPanelSummary)>
    <WithStyles(ExpansionPanelDetails)>
      <div>
        <WithStyles(FormControl)>
          <WithStyles(WithFormControlContext(InputLabel))
            htmlFor="deviceKey"
          >
            Select Channel
          </WithStyles(WithFormControlContext(InputLabel))>
          <WithStyles(WithFormControlContext(Select))
            inputProps={
              Object {
                "id": "channel",
                "name": "channel",
              }
            }
            onChange={[Function]}
          >
            <WithStyles(MenuItem)
              value="1"
            >
              33 (Dan's Raspberry)
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              value="2"
            >
              41 (Channel 41)
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              value="3"
            >
              42 (KCOP 15)
            </WithStyles(MenuItem)>
          </WithStyles(WithFormControlContext(Select))>
        </WithStyles(FormControl)>
        <WithStyles(FormControl)>
          <WithStyles(Input)
            id="input-with-icon-adornment"
            startAdornment={
              <WithStyles(WithFormControlContext(InputAdornment))
                position="start"
              >
                <pure(SearchIcon) />
              </WithStyles(WithFormControlContext(InputAdornment))>
            }
          />
        </WithStyles(FormControl)>
      </div>
      <WithStyles(EventsTable)
        data={Array []}
      />
    </WithStyles(ExpansionPanelDetails)>
  </WithStyles(ExpansionPanel)>
  <WithStyles(ExpansionPanel)
    expanded={false}
    onChange={[Function]}
  >
    <WithStyles(ExpansionPanelSummary)
      expandIcon={<pure(ExpandMoreIcon) />}
    >
      <WithStyles(Typography)>
        ADD A NEW EVENT
      </WithStyles(Typography)>
    </WithStyles(ExpansionPanelSummary)>
    <WithStyles(ExpansionPanelDetails)>
      <WithStyles(FormControl)>
        <WithStyles(WithFormControlContext(InputLabel))
          htmlFor="deviceKey"
        >
          Device
        </WithStyles(WithFormControlContext(InputLabel))>
        <WithStyles(WithFormControlContext(Select))
          inputProps={
            Object {
              "id": "deviceKey",
              "name": "deviceKey",
            }
          }
          onChange={[Function]}
          value=""
        />
      </WithStyles(FormControl)>
      <WithStyles(FormControl)
        aria-describedby="component-error-text"
        error={false}
        fullWidth={true}
      >
        <WithStyles(WithFormControlContext(InputLabel))
          htmlFor="component-error"
        >
          Data
        </WithStyles(WithFormControlContext(InputLabel))>
        <WithStyles(Input)
          name="data"
          onBlur={[Function]}
          onChange={[Function]}
          value=""
        />
        <WithStyles(WithFormControlContext(FormHelperText))
          id="component-error-text"
        >
          This field is required
        </WithStyles(WithFormControlContext(FormHelperText))>
      </WithStyles(FormControl)>
      <WithStyles(Button)
        color="secondary"
        onClick={[Function]}
        variant="contained"
      >
        Add Event
      </WithStyles(Button)>
    </WithStyles(ExpansionPanelDetails)>
  </WithStyles(ExpansionPanel)>
  <WithStyles(ConfirmDialog)
    contentText="A new event has been created"
    createButtonText="Create Event"
    onClickCreate={[Function]}
    onClickDismiss={[Function]}
    onClickView={[Function]}
    open={false}
    titleText="Confirm"
    viewButtonText="View Events"
  />
</div>
`;
